llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/Legality.h

```diff
@@ -167,7 +172,9 @@ class LegalityAnalysis {
   }
   /// Checks if it's legal to vectorize the instructions in \p Bndl.
   /// \Returns a LegalityResult object owned by LegalityAnalysis.
-  const LegalityResult &canVectorize(ArrayRef<Value *> Bndl);
+  /// p SkipScheduling skips the scheduler check and is only meant for testing.
```

Comment:
Also, this seems like kind of a weird flag to have for testing. If there is some non-scheduling legality check behavior we want to test in isolation maybe it could be refactored into a separate unit and tested there? But I'm a total newbie here, so maybe it's just me not having a good sense of what "legality checks minus scheduling" means. Feel free to ignore this comment if it doesn't make a lot of sense.

Reply:
We could split it into two separate functions, one that checks everything but the schedule, and a separate one for scheduling. But I would not like to make two functions available to the user, because legality should always do both. So using this flag seems to be the less bad thing to do API wise. We could also drop the flag completely once things become stable enough, or we could modify the tests such that they always pass the scheduling test.

Reply:
I added a TODO

Reply:
Yeah, writing the tests in a way that passes the scheduling checks would be one way to do it, or having two separate private functions and some (possibly hacky?) way to make them visible from tests only. Anyway, I think this is okay for now.

Reply:
Yeah, if we could test non-public functions then splitting them would be an option. There is some way of friending the tests, but anyway refactoring the tests shouldn't be too hard.